### DISCUSSION TITLE
How to get to that mythical 80% reduction?

### DISCUSSION THREAD

**Discussion by kevinrenskers** [firebase/firebase-js-sdk#4794] [Q&A] [answered]
https://firebase.google.com/docs/web/modular-upgrade:

> "Version 9 enables a dramatically reduced app size. It adopts the modern JavaScript Module format, allowing for "tree shaking" practices in which you import only the artifacts your app needs. Depending on your app, tree-shaking with version 9 can result in 80% less kilobytes than a comparable app built using version 8."

I'm wondering what kind of app could see a 80% reduction? I just refactored my JavaScript web-app from Firebase v8 to v9 beta 1, and the bundle size went from 220 KB transfer size to 166 KB. Of course that's a really great reduction, but not quite 80% haha (it's about 25%).

I use Authentication, Firestore, Storage and Functions, in a Svelte 3 app using Rollup.

**Comment by laurentpayot** [comment, marked as answer]
@kevinrenskers in the ["why change" page of the V9 docs](https://modularfirebase.web.app/get-started/why-change/), in the "The size difference" section, you’ll see that in the _very_ specific use case of only using Firestore to read data once the V9 is 80.06% lighter than the V8.
I agree it’s a _very very_ specific use case :wink:

**Reply by kevinrenskers** [reply]
I didn't even know about that website. I just used https://firebase.google.com/docs/firestore to guide my refactor. Looks like a nice site!
